### PR TITLE
dep: replace json-simple with gson

### LINKED ISSUE #2037: Upgrade dependencies to support Java 17
This issue is a part of the transition plan at https://github.com/iluwatar/java-design-patterns/wiki/19.-Java-17-transition. Here we upgrade the dependencies so that they support Java 17.

Acceptance criteria

- Project dependencies have been upgraded so that each of them supports Java 17

### PR DESCRIPTION
**Pull request by robertvolkmann**
Last release of `json-simple` is from 2009. 

Fixes #2037

### CODE REVIEW

File: typeobjectpattern/src/main/java/com/iluwatar/typeobject/JsonParser.java [hunk 1]
```diff
@@ -47,23 +43,21 @@ public class JsonParser {
...
-    var a = (JSONArray) jo.get("candies");
-    for (var o : a) {
-      var candy = (JSONObject) o;
-      var name = (String) candy.get("name");
-      var parentName = (String) candy.get("parent");
-      var t = (String) candy.get("type");
+  void parse() throws JsonParseException {
+    var is = this.getClass().getClassLoader().getResourceAsStream("candy.json");
+    var reader = new InputStreamReader(is);
+    var json = (JsonObject) com.google.gson.JsonParser.parseReader(reader);
```
Comment (AparnaData) [inline typeobjectpattern/src/main/java/com/iluwatar/typeobject/JsonParser.java:49]: Need to import this package com.google.gson.JsonParser.parseReader(reader)and can be used 
 JsonParser.parseReader(reader);

Comment (robertvolkmann) [reply typeobjectpattern/src/main/java/com/iluwatar/typeobject/JsonParser.java:49]: Works only if I rename the class `JsonParser`.

Comment (AparnaData) [reply typeobjectpattern/src/main/java/com/iluwatar/typeobject/JsonParser.java:49]: is there any other solution to fix it

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
> Last release of `json-simple` is from 2009.

Yeah, it was about the time to replace it 😅 Thanks for taking care of it!

**Pull request merged**